vscode/microsoft-kiota/src/commands/open-api-tree-view/search-or-open-api-description/searchOrOpenApiDescriptionCommand.ts

```diff
@@ -0,0 +1,67 @@
+import TelemetryReporter from "@vscode/extension-telemetry";
+import * as vscode from "vscode";
+
+import { extensionId, treeViewId } from "../../../constants";
+import { getExtensionSettings } from "../../../extensionSettings";
+import { setDeepLinkParams } from "../../../handlers/deepLinkParamsHandler";
+import { OpenApiTreeProvider } from "../../../openApiTreeProvider";
+import { searchSteps } from "../../../steps";
+import { IntegrationParams, validateDeepLinkQueryParams } from "../../../utilities/deep-linking";
+import { Command } from "../../Command";
+import { searchDescription } from "./searchDescription";
+
+export class SearchOrOpenApiDescriptionCommand extends Command {
+
+  private _openApiTreeProvider: OpenApiTreeProvider;
+  private _context: vscode.ExtensionContext;
+
+  constructor(openApiTreeProvider: OpenApiTreeProvider, context: vscode.ExtensionContext) {
+    super();
+    this._openApiTreeProvider = openApiTreeProvider;
+    this._context = context;
+  }
+
+  public getName(): string {
+    return `${treeViewId}.searchOrOpenApiDescription`;
+  }
+
+  public async execute(searchParams: Partial<IntegrationParams>): Promise<void> {
+    // set deeplink params if exists
+    if (Object.keys(searchParams).length > 0) {
+      let [params, errorsArray] = validateDeepLinkQueryParams(searchParams);
+      setDeepLinkParams(params);
+      const reporter = new TelemetryReporter(this._context.extension.packageJSON.telemetryInstrumentationKey);
+      reporter.sendTelemetryEvent("DeepLinked searchOrOpenApiDescription", {
+        "searchParameters": JSON.stringify(searchParams),
+        "validationErrors": errorsArray.join(", ")
+      });
+    }
+
+    // proceed to enable loading of openapi description
+    const yesAnswer = vscode.l10n.t("Yes, override it");
+    if (!this._openApiTreeProvider.isEmpty() && this._openApiTreeProvider.hasChanges()) {
```

Comment:
Would `this._openApiTreeProvider.hasChanges()` ever return true when `this._openApiTreeProvider.isEmpty()` is also true? If not, this can be:

```suggestion
    if (this._openApiTreeProvider.hasChanges()) {
```

Reply:
valid point. I see hasChanges first checks whether it's empty